### PR TITLE
[WIP][master] pylint check - standalone

### PR DESCRIPTION
**Pull request by fmessmer**
as suggested in #463 

I tried to implement this as a standalone script similar to `clang_format_check.sh` as suggested in https://github.com/ros-industrial/industrial_ci/pull/463#pullrequestreview-343986465
but I do not fully understand in what initial situation `pylint_check.sh` is called, i.e. what else needs to be done to setup target_workspace before `pylint` can start

### CODE REVIEW

File: industrial_ci/src/ci_main.sh [hunk 1]
```diff
@@ -50,6 +50,10 @@ elif [ -n "$CLANG_FORMAT_CHECK" ]; then
   # shellcheck source=industrial_ci/src/tests/clang_format_check.sh
   source "${ICI_SRC_PATH}/tests/clang_format_check.sh"
   run_clang_format_check
+elif [ -n "$PYLINT_CHECK" ]; then
```
Comment (ipa-mdl) [inline industrial_ci/src/ci_main.sh]:
```suggestion
elif [ "$PYLINT_CHECK" == true ]; then
```

File: industrial_ci/src/tests/pylint_check.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function run_pylint_check() {
+
+  ici_require_run_in_docker # this script must be run in docker
+  target_ws=~/target_ws
+
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/pylint_check.sh]:
```suggestion
  local -a sources
  ici_parse_env_array sources TARGET_WORKSPACE
  ici_run "prepare_sourcespace" ici_prepare_sourcespace "$target_ws" "${sources[@]}"

```

File: industrial_ci/src/tests/pylint_check.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  ici_run "prepare_sourcespace" ici_prepare_sourcespace "$target_ws" "${sources[@]}"
+
+  local -a pylint_versions
+  ici_parse_env_array pylint_versions PYLINT_VERSIONS
+  local -a pylint_args
+  ici_parse_env_array pylint_args PYLINT_ARGS
+
+  for cmd in "${pylint_versions[@]}"; do
+      ici_run "install_$cmd" ici_install_pkgs_for_command "$cmd" "$cmd"
+      ici_run "run_$cmd" "$cmd" "${pylint_args[@]}" "$(find "$target_ws/src" -iname "*.py")"
```
Comment (fmessmer) [inline industrial_ci/src/tests/pylint_check.sh]: seems like `target_ws` is incorrect:
```
find: ‘/root/target_ws/src’: No such file or directory
```
this worked in #463 - thus, there is still something missing...

Comment (ipa-mdl) [reply industrial_ci/src/tests/pylint_check.sh]: sry, my bad!

File: industrial_ci/src/tests/pylint_check.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function run_pylint_check() {
+  ici_require_run_in_docker # this script must be run in docker
+  target_ws=~/target_ws
+
+  local -a sources
+  ici_parse_env_array sources TARGET_WORKSPACE
+  ici_run "prepare_sourcespace" ici_prepare_sourcespace "$target_ws" "${sources[@]}"
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/pylint_check.sh]:
```suggestion
  ici_run "prepare_sourcespace" ici_prepare_sourcespace "$target_ws/src" "${sources[@]}"
```

**Comment by fmessmer** [comment]
seems like ROS modules cannot be found correctly, due to either not building the workspace or not being correctly sourced:
```
************* Module sample_talker
E:  3, 0: Unable to import 'std_msgs.msg' (import-error)
```

I tried `ici_with_ws "$target_ws" ici_run "run_$cmd" "$cmd" "${pylint_args[@]}" "$(find "$TARGET_REPO_PATH" -iname "*.py")"` and `rosenv`, but without much luck...

**Comment by ipa-mdl** [comment]
> seems like ROS modules cannot be found correctly, due to either not being build or not being correctly sourced:

Okay, so stand-alone mode is not an option.
Thanks for trying it out!
We should proceed with #463

**Comment by fmessmer** [comment]
> > seems like ROS modules cannot be found correctly, due to either not being build or not being correctly sourced:
> 
> Okay, so stand-alone mode is not an option.
> Thanks for trying it out!
> We should proceed with #463

I'm not yet sure whether it's not possible or just not implemented correctly...I had similar issues with #463, too

**Comment by ipa-mdl** [comment]
> I'm not yet sure whether it's not possible or just not implemented correctly

If the dependencies must be present for `pylint` to succeed, then it should go into `source_test.sh`.

**Comment by ipa-mdl** [comment]
> due to either not being build or not being correctly sourced:

As a stand-alone test, nothing gets built and nothing can be sourced.

**Comment by fmessmer** [comment]
doesn't work standalone without sourcing...thus closing in favor of #463

**Pull request closed without merging**